### PR TITLE
inital prep for v3.

### PR DESCRIPTION
**Pull request by siennathesane**
This is the base branch to make progress on the v3 work and start getting things in place.

Signed-off-by: Mike Lloyd <mike@reboot3times.org>

### CODE REVIEW

**Comment by mxplusb** [comment]
@janisz can you take a look at the build problems for me? My knowledge of go modules is weak and I can't figure out the problem.

File: go.mod [hunk 1]
```diff
@@ -1,3 +1,5 @@
-module github.com/allegro/bigcache/v2
+module github.com/allegro/bigcache/v3
 
 go 1.12
+
+require github.com/allegro/bigcache/v3 v3.0.0
```
Comment (janisz) [inline go.mod:5]: I think this is not needed

File: go.sum [hunk 2]
```diff
@@ -0,0 +1,3 @@
+github.com/allegro/bigcache v1.2.1 h1:hg1sY1raCwic3Vnsvje6TT7/pnZba83LeFck5NrFKSc=
+github.com/allegro/bigcache/v2 v2.2.5 h1:mRc8r6GQjuJsmSKQNPsR5jQVXc8IJ1xsW5YXUYMLfqI=
+github.com/allegro/bigcache/v2 v2.2.5/go.mod h1:FppZsIO+IZk7gCuj5FiIDHGygD9xvWQcqg1uIPMb6tY=
```
Comment (janisz) [inline go.sum:3]: I think this is not needed

**Comment by janisz** [comment]
```diff
diff --git a/examples_test.go b/examples_test.go
index f447072..ccc513a 100644
--- a/examples_test.go
+++ b/examples_test.go
@@ -5,7 +5,7 @@ import (
        "log"
        "time"
 
-       "github.com/allegro/bigcache/v2"
+       "github.com/allegro/bigcache/v3"
 )
 
 func Example() {
diff --git a/go.mod b/go.mod
index 6f045f2..8be47fa 100644
--- a/go.mod
+++ b/go.mod
@@ -1,5 +1,3 @@
 module github.com/allegro/bigcache/v3
 
 go 1.12
-
-require github.com/allegro/bigcache/v2 v2.2.5
diff --git a/go.sum b/go.sum
index fdc2c46..e69de29 100644
--- a/go.sum
+++ b/go.sum
@@ -1,3 +0,0 @@
-github.com/allegro/bigcache v1.2.1 h1:hg1sY1raCwic3Vnsvje6TT7/pnZba83LeFck5NrFKSc=
-github.com/allegro/bigcache/v2 v2.2.5 h1:mRc8r6GQjuJsmSKQNPsR5jQVXc8IJ1xsW5YXUYMLfqI=
-github.com/allegro/bigcache/v2 v2.2.5/go.mod h1:FppZsIO+IZk7gCuj5FiIDHGygD9xvWQcqg1uIPMb6tY=
```

**Comment by janisz** [comment]
It looks like we have a flaky test `TestWriteAndReadParallelSameKeyWithStats`

**Comment by mxplusb** [comment]
@cristaloleg and @janisz I'm working on the interface definition for #204 via 34853d0, do we want to cut 3.0 with just the interface then add in a new default implementation in 3.1? I want to get the interface definition with 3.0, but I don't want the release to be held back on the initial implementation since this fixes #148.

**Comment by janisz** [comment]
IMO we should release this and add new features in feature release. 

Still I'm not sure if we lost backward compatibility. Do we really need 3.0 release?

**Comment by mxplusb** [comment]
> I'm not sure if we lost backward compatibility

I'm not either but better to be safe than sorry.

**Pull request closed without merging**